types/datasource.d.ts

```diff
@@ -76,28 +90,238 @@ import {IsolationLevel, Transaction} from './transaction-mixin';
  */
 export declare class DataSource extends EventEmitter {
   name: string;
-  settings: Options;
+  settings: ConnectorSettings;
 
   initialized?: boolean;
   connected?: boolean;
   connecting?: boolean;
+  
+  /**
+   * {@inheritDoc DataSource.connector}
+   * @deprecated Use {@link DataSource.connector} instead.
+   */
+  adapter?: Connector;
 
+  /**
+   * Connector instance.
+   */
   connector?: Connector;
 
-  definitions: {[modelName: string]: ModelDefinition};
+  modelBuilder: ModelBuilder;
 
+  models: Record<string, typeof ModelBase>;
+  
+  definitions: {[modelName: string]: ModelDefinition};
```

Comment:
Will need to check and document these' relationship with Connector._models

Reply:
These are populated from the same-named properties of the ModelBuilder instance passed during class initialisation.